tests/SimplifyTest.elm

```diff
@@ -673,6 +674,132 @@ a =
 
 
 
+-- LAMBDA REDUCE
+
+
+lambdaReduceTests : Test
+lambdaReduceTests =
+    describe "lambda reduce"
+        [ test "should detect (\\error -> Err error) as Err function" <|
+            \() ->
+                """module A exposing (..)
+a = Result.mapError f << (\\error -> Err error)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "Using Result.mapError on Err will result in Err with the function applied to the error"
+                            , details = [ "You can replace this call by Err with the function directly applied to the error itself." ]
+                            , under = "Result.mapError"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = Err << f
```

Comment:
The only part I'm unsure about is as to whether we should keep the user's original code for the error function. But... I think this is going to be okay in practice?